src/rust/engine/src/downloads.rs

```diff
@@ -26,30 +28,40 @@ struct NetDownload {
 
 impl NetDownload {
   async fn start(core: &Arc<Core>, url: Url, file_name: String) -> Result<NetDownload, String> {
-    // TODO: Retry failures
-    let response = core
+    let try_download = || async {
+      core
       .http_client
       .get(url.clone())
       .send()
       .await
-      .map_err(|err| format!("Error downloading file: {}", err))?;
+      .map_err(|err| (format!("Error downloading file: {}", err), true))
+      .and_then(|res|
+        // Handle common HTTP errors.
+        if res.status().is_server_error() {
+          Err((format!(
+            "Server error ({}) downloading file {} from {}",
+            res.status().as_str(),
+            file_name,
+            url,
+          ), true))
+        } else if res.status().is_client_error() {
+          Err((format!(
+            "Client error ({}) downloading file {} from {}",
+            res.status().as_str(),
+            file_name,
+            url,
+          ), false))
+        } else {
+          Ok(res)
+        })
+    };
+
+    // TODO: Allow the retry strategy to be configurable?
+    let retry_strategy = ExponentialBackoff::from_millis(10).map(jitter).take(3);
```

Comment:
```suggestion
    let retry_strategy = ExponentialBackoff::from_millis(10).map(jitter).take(4);
```
Max of 1 second backoff is probably sufficient for a local datacenter, but probably not for the open internet. Let's do 10 seconds.